### PR TITLE
[FLINK-14849][hive][doc] Fix documentation about Hive dependencies

### PR DESCRIPTION
**Pull request by lirui-apache**
## What is the purpose of the change

Fix the documentation about dependencies needed to use Hive connector.

## Brief change log

  - Update documentation about Hive dependencies

## Verifying this change

NA

## Does this pull request potentially affect one of the following parts:

 NA

## Documentation

NA

### CODE REVIEW

**Comment by lirui-apache** [comment]
The idea is to let users add required dependencies at runtime, e.g. by putting the jars in the `lib` folder. And if users are using Hive connector programmatically, they don't need to pack any of these dependencies in their jar file, in order to reduce the chance of lib conflicts.
I also manually identified the required deps for each Hive major version and listed them in the doc.
@bowenli86 @JingsongLi Please have a look. Thanks.

**Comment by lirui-apache** [comment]
I'll also update the `zh` docs once the changes are finalized.

**Comment by JingsongLi** [comment]
Thanks @lirui-apache for the great works!

**Comment by JingsongLi** [comment]
I think this is pretty good, can you build web-site and show the picture?

**Comment by lirui-apache** [comment]
@JingsongLi  Attached are screenshots of my locally built web-site.

**Comment by JingsongLi** [comment]
LGTM

File: docs/dev/table/hive/index.md [hunk 1]
```diff
@@ -100,147 +102,203 @@ We are using Hive 2.3.4 and 1.2.1 as examples here.
...
+       // you can use your own hadoop jars. Either way, make sure it's compatible with your Hadoop
+       // cluster and the Hive version you're using.
+       flink-shaded-hadoop-2-uber-2.7.5-{{ site.shaded_version }}.jar
+
+       // Hive dependencies
+       hive-exec-2.2.0.jar
+
+       // Orc dependencies -- required by the ORC vectorized optimizations
+       orc-core-1.4.3.jar
+       aircompressor-0.8.jar
```
Comment (bowenli86) [inline docs/dev/table/hive/index.md]: I didn't see this in our pom file?

File: docs/dev/table/hive/index.md [hunk 2]
```diff
@@ -100,147 +102,203 @@ We are using Hive 2.3.4 and 1.2.1 as examples here.
...
+
+       // Hadoop dependencies
+       // You can pick a pre-built Hadoop uber jar provided by Flink, alternatively
+       // you can use your own hadoop jars. Either way, make sure it's compatible with your Hadoop
+       // cluster and the Hive version you're using.
+       flink-shaded-hadoop-2-uber-2.8.3-{{ site.shaded_version }}.jar
+
+       // Hive dependencies
+       hive-exec-3.1.0.jar
+       libfb303-0.9.3.jar
```
Comment (bowenli86) [inline docs/dev/table/hive/index.md]: also didn't see this in our pom file?

File: docs/dev/table/hive/index.md (hunk 1 above)
Comment (lirui-apache) [reply docs/dev/table/hive/index.md]: We depend on `orc-core` in our `hive-2.2.0` profile. `aircompressor` is a transitive dependency of `orc-core`

File: docs/dev/table/hive/index.md (hunk 2 above)
Comment (lirui-apache) [reply docs/dev/table/hive/index.md]: `libfb303` is either packed into `hive-exec` or introduced as a transitive dependency of `hive-exec`. So we don't need to declare it in our pom

**Comment by lirui-apache** [comment]
@bowenli86 I have sync'ed the changes to zh doc.

File: docs/dev/table/hive/index.md (hunk 1 above)
Comment (bowenli86) [reply docs/dev/table/hive/index.md]: can we add this explanation to documentation? like 
```suggestion
       aircompressor-0.8.jar  // transitive dependency of orc-core
```

File: docs/dev/table/hive/index.md (hunk 2 above)
Comment (bowenli86) [reply docs/dev/table/hive/index.md]: can we add this explanation to documentation?

**Comment by lirui-apache** [comment]
@bowenli86 PR updated to add the explanations.

**Pull request closed without merging**